src/main/java/org/junitpioneer/jupiter/ExpectedToFailExtension.java

```diff
@@ -42,7 +42,12 @@ private static void invokeAndInvertResult(Invocation<Void> invocation, Extension
 				throw t;
 			}
 
-			String message = getExpectedToFailAnnotation(extensionContext).value();
+			ExpectedToFail expectedToFail = getExpectedToFailAnnotation(extensionContext);
+			if (!expectedToFail.onExceptions()[0].isInstance(t)) {
+				fail("Test marked as 'expected to fail' failed with an unexpected " + t.getClass() + " exception");
```

Comment:
`Assertions::fail` can take a second parameter `Throwable` cause, which I think is applicable here?

Reply:
So you mean that if the test throws an `AssertionFailedError` then we should check its cause (if any) against the `onExceptions` types?

Reply:
Hm, no, I don't think that's right. Assertion errors are assertion failures which is not the use-case I had in mind... Unless you can justify it?

I just meant we should not be "swallowing" unexpected exceptions because it makes debugging harder.

Reply:
OK, I see what you mean. That of course makes sense.